src/Components/Directory.Build.targets

```diff
@@ -30,4 +30,94 @@
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- ensure the config generator is built -->
+    <ProjectReference Include="$(ConfigurationSchemaGeneratorProjectPath)"
+                      Private="false"
+                      ReferenceOutputAssembly="false" />
+  </ItemGroup>
+
+  <Target Name="GenerateConfigurationSchema"
+          DependsOnTargets="CoreGenerateConfigurationSchema;
+                            CopyNewConfigurationSchema;
+                            CompareConfigurationSchema" />
+
+  <Target Name="WriteConfigurationSchemaGeneratorRspFile">
+    <PropertyGroup>
+      <ConfigSchemaInput>"@(IntermediateAssembly)"</ConfigSchemaInput>
+      <ConfigSchemaReferences>"@(ReferencePathWithRefAssemblies, '" "')"</ConfigSchemaReferences>
+      <ConfigSchemaOutput>"$(GeneratedConfigurationSchemaOutputPath)"</ConfigSchemaOutput>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ConfigSchemaGenArg Include="--input $(ConfigSchemaInput)" />
+      <ConfigSchemaGenArg Include="--reference $(ConfigSchemaReferences)" />
+      <ConfigSchemaGenArg Include="--output $(ConfigSchemaOutput)" />
+    </ItemGroup>
+
+    <WriteLinesToFile File="$(ConfigurationSchemaGeneratorRspPath)"
+                      Lines="@(ConfigSchemaGenArg)"
+                      Overwrite="true"
+                      WriteOnlyWhenDifferent="true" />
+  </Target>
+
+  <Target Name="CalculateConfigurationSchemaGeneratorPath">
+    <MSBuild Projects="$(ConfigurationSchemaGeneratorProjectPath)"
+             Targets="GetTargetPath">
+      <Output TaskParameter="TargetOutputs" PropertyName="ConfigurationSchemaGeneratorPath" />
+    </MSBuild>
+  </Target>
+
+  <Target Name="CoreGenerateConfigurationSchema"
+          DependsOnTargets="WriteConfigurationSchemaGeneratorRspFile;
+                            CalculateConfigurationSchemaGeneratorPath"
+          Inputs="$(ConfigurationSchemaGeneratorPath);
+                  @(IntermediateAssembly);
+                  @(ReferencePathWithRefAssemblies)"
+          Outputs="$(GeneratedConfigurationSchemaOutputPath)">
+    <PropertyGroup>
+
+      <GeneratorCommandLine>"$(DotNetTool)" exec $(ConfigurationSchemaGeneratorPath)</GeneratorCommandLine>
+      <GeneratorCommandLine>$(GeneratorCommandLine) @$(ConfigurationSchemaGeneratorRspPath)</GeneratorCommandLine>
+    </PropertyGroup>
+
+    <Exec Command="$(GeneratorCommandLine)" />
+
+    <ItemGroup>
+      <FileWrites Include="$(GeneratedConfigurationSchemaOutputPath)" Condition="Exists('$(GeneratedConfigurationSchemaOutputPath)')"/>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CopyNewConfigurationSchema"
+          Condition="'$(ConfigurationSchemaExists)' != 'true' OR '$(UpdateConfigurationSchema)' == 'true'">
+
+    <Copy SourceFiles="$(GeneratedConfigurationSchemaOutputPath)"
+          DestinationFiles="$(ConfigurationSchemaPath)" />
+
+  </Target>
+
+  <Target Name="CompareConfigurationSchema"
```

Comment:
Is it supposed to be where the logic to verify the json file that is committed in the repos is up to date with the curent code?

Reply:
Correct.